pydantic/decorators.py

```diff
@@ -116,21 +117,51 @@ def __call__(self, __cls: Any, __values: _decorators.RootValidatorValues) -> _de
 )
 
 
+# Use a single overload here to get a different signature that requires at least one field be specified
+@overload  # type: ignore[misc]
 def validator(
     __field: str,
     *fields: str,
     pre: bool = False,
+    each_item: bool = False,
+    always: bool = False,
     check_fields: bool | None = None,
     allow_reuse: bool = False,
+) -> Callable[[_V1ValidatorType], _V1ValidatorType]:
+    ...
+
+
+def validator(
+    *fields: str,
+    pre: bool = False,
     each_item: bool = False,
+    always: bool = False,
+    check_fields: bool | None = None,
+    allow_reuse: bool = False,
 ) -> Callable[[_V1ValidatorType], _V1ValidatorType]:
     """
     Decorate methods on the class indicating that they should be used to validate fields
     :param fields: which field(s) the method should be called on
+    :param pre: whether or not this validator should be called before the standard validators (else after)
+    :param each_item: for complex objects (sets, lists etc.) whether to validate individual elements rather than the
+      whole object
+    :param always: whether this method and other validators should be called even if the value is missing
     :param check_fields: whether to check that the fields actually exist on the model
     :param allow_reuse: whether to track and raise an error if another validator refers to the decorated function
     """
-    fields = tuple((__field, *fields))
+    if not fields:
```

Comment:
I re-added all these errors, mostly because I ended up getting surprising behavior while dev'ing on this because I did something that was type-invalid but it didn't raise any runtime errors and just silently didn't validate anything.

Considering this is only executed during the decorator call (not during actual validation) I feel like these checks (and error messages at point-of-failure) are worth including. I've also added them to `field_validator`, and re-added the v1 tests for this behavior that had been removed.

I am happy to undo this if preferred, but it would have helped me during development if they were there.

Reply:
Also, I left them as `TypeError` for now; I'll convert them into some form of PydanticError with proper error codes etc. as part of that work.

Reply:
Personally I think these are fine as TypeError. They should be quite self explanatory.

Reply:
I think worth use `PydanticTypeError` down the road.